tsdapilib/backends.py

```diff
@@ -0,0 +1,265 @@
+
+import psycopg2
+import psycopg2.extensions
+import psycopg2.pool
+import sqlite3
+
+from collections import OrderedDict
+from contextlib import contextmanager
+from datetime import datetime, timedelta
+from typing import Optional, ContextManager, Any
+
+
+class GenericBackend(object):
```

Comment:
ABC?

Reply:
Did think about it, but since I actually implement a few methods I opted not to use ABC

Reply:
I don't think this is a show-stopper, unless I've been misusing the ABC concept all along (which might very well be the case). Implemented functions will still be inherited, direct instancing is blocked, and you get to use abstract method decorators for the unimplemented bits.

Reply:
Ah yes, I've always thought of ABCs as "type definitions" for classes